src/phoenix/server/app.py

```diff
@@ -104,6 +109,37 @@ async def get(self, request: Request) -> FileResponse:
         )
 
 
+T = TypeVar("T", contravariant=True)
+
+
+class SupportsPut(Protocol[T]):
+    def put(self, obj: T) -> None:
+        ...
```

Comment:
The generic for the one case seems strange. Why not explicitly type it as the queue?